File: examples/exampleHollowCylinder.m
clc
clear

DRAW_MAGNET = 1;
DRAW_TIKZ   = 0;

%% Define cross sections

hollowCylinder1 = CrossSectHollowCylinder( ...
        'name', 'hollowCylinder1', ...
        'dim_t', DimMillimeter(4), ...
        'dim_r_o', DimMillimeter(80), ...
        'location', Location2D( ...
            'anchor_xy', DimMillimeter([0,0]), ...
            'theta', DimDegree(0).toRadians() ...
        ) ...
        );

%% Define components

cs = hollowCylinder1;

comp1 = Component( ...
        'name', 'comp1', ...
        'crossSections', cs, ...
        'material', MaterialGeneric('name', 'pm'), ...
        'makeSolid', MakeExtrude( ...
            'location', Location3D( ...
                'anchor_xyz', DimMillimeter([0,0,0]), ...
                'rotate_xyz', DimDegree([0,0,0]).toRadians() ...
                ), ...
            'dim_depth', DimMillimeter(80)) ...
        );

%% Draw via MagNet

if (DRAW_MAGNET)
    toolMn = MagNet();
    toolMn.open();
    toolMn.setVisibility(1);
    toolMn.setDefaultLengthUnit('DimMillimeter', false);

    comp1.make(toolMn, toolMn);

    toolMn.viewAll();
end

%% Draw via TikZ

if (DRAW_TIKZ)
    toolTikz = TikZ();
    toolTikz.open('output.txt');

    comp1.draw(toolTikz);

    toolTikz.close();
end

File: examples/mach_cad_examples/example_hollow_cylinder.py
import os
import sys

# change current working directory to file location
os.chdir(os.path.dirname(__file__))
# add the directory immediately above this file's directory to path for module import
sys.path.append("../..")

import mach_cad.tools.magnet as mn
import mach_cad.model_obj as mo

x = mo.DimMillimeter(4)
y = mo.DimMillimeter(80)
z = mo.DimMillimeter(40)

# create hollowcylinder crossection object
hollowCylinder1 = mo.CrossSectHollowCylinder(
    name="hollowCylinder1", location=mo.Location2D(), dim_t=x, dim_r_o=y
)

# create hollowcylinder component
comp1 = mo.Component(
    name="comp1",
    cross_sections=[hollowCylinder1],
    material=mo.MaterialGeneric(name="pm"),
    make_solid=mo.MakeExtrude(location=mo.Location3D(), dim_depth=x),
)

# create an instance of the MagNet class
toolMn = mn.MagNet(visible=True)
toolMn.open()

comp1.make(toolMn, toolMn)
toolMn.view_all()

